app/kuma-dp/cmd/run.go

```diff
@@ -100,13 +100,17 @@ func newRunCmd() *cobra.Command {
 				runLog.Info("generated Envoy configuration will be stored in a temporary directory", "dir", tmpDir)
 			}
 
-			dataplane := envoy.New(envoy.Opts{
+			dataplane, err := envoy.New(envoy.Opts{
 				Catalog:   catalog,
 				Config:    cfg,
 				Generator: bootstrapGenerator,
 				Stdout:    cmd.OutOrStdout(),
 				Stderr:    cmd.OutOrStderr(),
 			})
+			if err != nil {
+				runLog.Error(err, `could not find the "envoy" executable in your path`)
```

Comment:
What happens if `envoy` is found but running the executable fails for some reasons (ie, corrupted Envoy executable)? I would like to implement two separate checks:

* First determine if `envoy` exists as a file before executing it, if not return `could not find the "envoy" executable in your path`.
* Then, if found, when executing `envoy` if it fails for some reason return a generic error `the "envoy" executable was found at "/path" but an error occurred when executing it`.